packages/js/product-editor/src/hooks/use-product-metadata.ts

```diff
@@ -2,28 +2,58 @@
  * External dependencies
  */
 import { useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { Product } from '@woocommerce/data';
 /**
  * Internal dependencies
  */
 import { Metadata } from '../types';
 
-function useProductMetadata( postType?: string ) {
-	const [ metadata, setMetadata ] = useEntityProp< Metadata< string >[] >(
+function useProductMetadata(
+	postType?: string,
+	id?: number,
+	parentPostType?: string
+) {
+	const [ thisMetadata, setMetadata ] = useEntityProp< Metadata< string >[] >(
 		'postType',
 		postType || 'product',
 		'meta_data'
 	);
 
+	const metadata = useSelect(
+		( select ) => {
+			let usedMetadata: Metadata< string >[];
+			if ( id ) {
+				// @ts-expect-error There are no types for this.
+				const { getEditedEntityRecord } = select( 'core' );
+				const { meta_data: parentMetadata }: Product =
+					getEditedEntityRecord( 'postType', parentPostType, id );
+				usedMetadata = parentMetadata;
+			} else {
+				usedMetadata = thisMetadata;
```

Comment:
So if the ID `id` is not defined, the selector returns `thisMetadata`. Why not handle selecting the data from this selector? You could pick the post `id` either from the context by using the `useEntityId()` or from the hook param when it's defined.